media/linux/pds-sqlite3-queries/create-training-rosters.py

```diff
@@ -75,134 +109,53 @@ def _dt_to_int(datetime):
             continue
 
         for req in m[key]:
-            if(req['description'] != training_to_find):
+            if(req['description'] != training_to_find['pds_type']):
                 continue
             reqcount += 1
             mem = pretty_member(m)
```

Comment:
A thought: rather than making a whole new dictionary with just the fields you want from the member, why not just have `pretty_member()` add a `['calculated']` field to the member with the things that you had to calculate?  That way, you don't have yet another object being passed around, and you're not needlessly copying member data multiple times.

The PDS member dictionary is not sacrosanct; if you need to do a little processing on the member data, you can cache the results on the member dictionary itself, and then just pass the member itself around (without making another new object/dictionary/whatever).  This would prevent you needing to copy all the base data from the member (e.g., the MID, the name, etc.).

Reply:
It looks like you have cut down on the redundant information; good!

I'd suggest renaming `m` to be `member` (one of Python's goals is to be clear / easy to read code -- and in most of the rest of ECC Python, I've tried to be consistent to use the name `member` just to be simple and to the point).  And also rename `mem` to be something else -- this is *not* the member (as the name `mem` implies) -- it's calculated metadata about the member.  Just cache it on the member itself somewhere.

`out[mid][sd]` could probably become shorter/simpler, too...?  (I haven't thought this through, but it seems possible)